saleor/graphql/product/mutations/products.py

```diff
@@ -846,3 +851,161 @@ def mutate(cls, root, info, image_id, variant_id):
                     variant_image.delete()
         return VariantImageUnassign(
             product_variant=variant, image=image, errors=errors)
+
+class ProductVariantDigitalInput(graphene.InputObjectType):
+    max_downloads = graphene.Int(
+        description='[Optional] Maximal number of downloading for one url',
+        required=False, name='max_downloads')
+    url_valid_days = graphene.Int(
+        description='[Optional] Number of days which each url will be active',
+        required=False, name='url_valid_days'
+    )
+    automatic_fulfillment = graphene.Boolean(
+        description='Overwrite default automatic_fulfillment setting for variant',
+        required=False, name='automatic_fulfillment'
+    )
+
+
+class ProductVariantDigitalUploadInput(ProductVariantDigitalInput):
+    content_file = Upload(
+        required=True,
+        description='Represents an file in a multipart request.')
+
+
+class ProductVariantDigitalUpload(BaseMutation):
+    variant = graphene.Field(ProductVariant)
+    content = graphene.Field(DigitalContent)
+
+    class Arguments:
+        id = graphene.ID(
+            description='ID of a product variant to upload digital content.',
+            required=True)
+        input = ProductVariantDigitalUploadInput(
+            required=True,
+            description='Fields required to create a digital content.')
+
+    class Meta:
+        description = dedent('''Create new digital content. This mutation must 
+        be sent as a `multipart` request. More detailed specs of the upload 
+        format can be found here:
+        https://github.com/jaydenseric/graphql-multipart-request-spec''')
+
+    @classmethod
+    @permission_required('product.manage_products')
+    def mutate(cls, root, info, id, input):
+        errors = []
+        variant = cls.get_node_or_error(
+            info, id, errors, 'id', only_type=ProductVariant)
+        content_data = info.context.FILES.get(input['content_file'])
+        if hasattr(variant, 'digital_content'):
```

Comment:
As we spoke - in this case, we could simply override existing content instead of raising an error.